GVFS/GVFS.Common/ModifiedPathsDatabase.cs

```diff
@@ -183,6 +185,22 @@ private bool TryParseRemoveLine(string line, out string key, out string error)
             return true;
         }
 
+        private bool ContainsParentDirectory(string modifiedPath)
+        {
+            string[] pathParts = modifiedPath.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+            StringBuilder parentFolder = new StringBuilder();
+            foreach (string pathPart in pathParts.Take(pathParts.Length - 1))

```

Comment:
Not sure how much of a perf impact this has, but `Take` is causing us to construct a new object, whereas we could just use a for loop here and the code would look very similar. I doubt this matters at all when checking a single entry, but it could add up for someone who has 100's of thousands of entries.